lib/Graph/Graph.cpp

```diff
@@ -649,6 +649,37 @@ TransposeNode *Function::createTranspose(llvm::StringRef name, NodeValue input,
   return addNode(new TransposeNode(name, NT, input, shuffle.vec()));
 }
 
+static void mergeMultidirectionalBroadcast(std::vector<size_t> &mergeShape,
+                                           llvm::ArrayRef<size_t> shape) {
+  size_t shift = mergeShape.size() - shape.size();
+  for (size_t i = 0; i < shape.size(); i++) {
+    if (shape[i] != 1) {
+      assert(((shape[i] == mergeShape[shift + i]) ||
+              (mergeShape[shift + i] == 1)) &&
+             "Incompatible dimension for the broadcast");
+      mergeShape[shift + i] = shape[i];
+    }
+    // Otherwise, just leave mergeShape[i] as it is
+  }
+}
+
+std::vector<size_t>
+Function::computeMultidirectionalBroadcast(llvm::ArrayRef<size_t> shape0,
```

Comment:
Why does this function have to be here in Function? It's only used in the loader.

Reply:
Ah, good catch @jackm321